File: apps/vertx-service/src/main/java/com/myapp/helper/BaseResponse.java
package helper;

import io.vertx.core.json.JsonObject;
import io.vertx.ext.web.RoutingContext;

public class BaseResponse {

    public static void sendResponse(RoutingContext context, String resMessage, String responseType, Object result) {
        JsonObject response;
        int status;

        switch (responseType) {
            case "created":
                response = CreatedResponse(resMessage);
                status = 201;
                break;
            case "success":
                response = SuccessResponse(result, resMessage);
                status = 200;
                break;
            case "unauthorized":
            case "forbidden":
                response = UnauthorizedResponse(resMessage);
                status = 403;
                break;
            case "internalServerError":
                response = InternalServerErrorResponse();
                status = 500;
                break;
            default:
                response = SuccessResponse(result, resMessage);
                status = 200;
        }

        // Set status and send response
        context.response()
            .setStatusCode(status)
            .putHeader("content-type", "application/json")
            .end(response.encode());
    }

    private static JsonObject CreatedResponse(String message) {
        return new JsonObject()
            .put("status", "created")
            .put("message", message);
    }

    private static JsonObject SuccessResponse(Object result, String message) {
        return new JsonObject()
            .put("status", "success")
            .put("message", message)
            .put("data", result);
    }

    private static JsonObject UnauthorizedResponse(String message) {
        return new JsonObject()
            .put("status", "unauthorized")
            .put("message", message);
    }

    private static JsonObject InternalServerErrorResponse() {
        return new JsonObject()
            .put("status", "error")
            .put("message", "Internal Server Error");
    }
}

File: apps/actix-service/src/helper/base_response.rs
use actix_web::{HttpResponse, Responder};
use serde_json::json;

pub struct BaseResponse;

impl BaseResponse {
    pub fn send_response(
        response_type: &str,
        res_message: &str,
        result: Option<serde_json::Value>,
    ) -> impl Responder {
        match response_type {
            "created" => Self::created_response(res_message),
            "success" => Self::success_response(result, res_message),
            "unauthorized" | "forbidden" => Self::unauthorized_response(res_message),
            "internalServerError" => Self::internal_server_error_response(),
            _ => Self::success_response(result, res_message),
        }
    }

    fn created_response(message: &str) -> HttpResponse {
        HttpResponse::Created()
            .content_type("application/json")
            .json(json!({
                "status": "created",
                "message": message
            }))
    }

    fn success_response(result: Option<serde_json::Value>, message: &str) -> HttpResponse {
        HttpResponse::Ok()
            .content_type("application/json")
            .json(json!({
                "status": "success",
                "message": message,
                "data": result.unwrap_or(json!(null))
            }))
    }

    fn unauthorized_response(message: &str) -> HttpResponse {
        HttpResponse::Forbidden()
            .content_type("application/json")
            .json(json!({
                "status": "unauthorized",
                "message": message
            }))
    }

    fn internal_server_error_response() -> HttpResponse {
        HttpResponse::InternalServerError()
            .content_type("application/json")
            .json(json!({
                "status": "error",
                "message": "Internal Server Error"
            }))
    }
}
